pkg/daemon/daemon.go

```diff
@@ -498,60 +505,51 @@ func (dn *Daemon) triggerUpdate() error {
 	return dn.triggerUpdateWithMachineConfig(nil)
 }
 
-// isDesiredMachineState confirms that the node is actually in the state that it
-// wants to be in. It does this by looking at the elements in the target config
-// and checks if all are present on the node. Returns true iff there are no
-// mismatches (e.g. files, units, OS version), as well as the config that was
-// evaluated if the state is what the machine wants to be in.
-func (dn *Daemon) isDesiredMachineState() (bool, string, error) {
+// validateOnDiskState compares the on-disk state against what a configuration
+// specifies.  If for example an admin ssh'd into a node, or another operator
+// is stomping on our files, we want to highlight that and mark the system
+// degraded.
+func (dn *Daemon) validateOnDiskState(currentConfig *mcfgv1.MachineConfig) bool {
+	if !dn.checkFiles(currentConfig.Spec.Config.Storage.Files) {
+		return false
+	}
+	if !dn.checkUnits(currentConfig.Spec.Config.Systemd.Units) {
+		return false
+	}
+	return true
+}
+
+// isDesiredMachineState looks at the node annotations and determines if
+// our current state matches the desired one.
+func (dn *Daemon) isDesiredMachineState() (bool, *mcfgv1.MachineConfig, error) {
 	ccAnnotation, err := getNodeAnnotation(dn.kubeClient.CoreV1().Nodes(), dn.name, CurrentMachineConfigAnnotationKey)
 	if err != nil {
-		return false, "", err
+		return false, nil, err
 	}
 	dcAnnotation, err := getNodeAnnotation(dn.kubeClient.CoreV1().Nodes(), dn.name, DesiredMachineConfigAnnotationKey)
 	if err != nil {
-		return false, "", err
+		return false, nil, err
 	}
 
 	currentConfig, err := getMachineConfig(dn.client.MachineconfigurationV1().MachineConfigs(), ccAnnotation)
 	if err != nil {
-		return false, "", err
+		return false, nil, err
 	}
+	currentConfigName := currentConfig.GetName()
 	desiredConfig, err := getMachineConfig(dn.client.MachineconfigurationV1().MachineConfigs(), dcAnnotation)
 	if err != nil {
-		return false, "", err
-	}
-
-	// if we can't reconcile the changes between the old config and the new
-	// config, the machine is definitely not in its desired state. this function
-	// will return true (meaning they are reconcilable) if there aren't actually
-	// changes.
-	reconcilableError := dn.reconcilable(currentConfig, desiredConfig)
-	if reconcilableError != nil {
-		return false, "", nil
+		return false, nil, err
 	}
+	desiredConfigName := desiredConfig.GetName()
 
-	isDesiredOS := false
-	// We only deal with operating system management on RHCOS
-	if dn.OperatingSystem != MachineConfigDaemonOSRHCOS {
```

Comment:
is all checking for OS version betting dropped?

Reply:
Nope, we still have:
```
func (dn *Daemon) updateOS(oldConfig, newConfig *mcfgv1.MachineConfig) error {
	if dn.OperatingSystem != MachineConfigDaemonOSRHCOS {
		glog.V(2).Infof("Updating of non RHCOS nodes are not supported")
		return nil
	}
```

In the big picture, this is dropping a lot of duplicate logic between "check for update" and "update".  Now "check for update" is just "compare machine config names".